Userland/Libraries/LibWasm/WASI/Wasi.cpp

```diff
@@ -588,7 +588,8 @@ ErrorOr<Result<void>> Implementation::impl$path_create_directory(Configuration&
     auto slice = TRY(slice_typed_memory(configuration, path, path_len));
     auto null_terminated_string = DeprecatedString::copy(slice);
 
-    if (mkdirat(dir_fd, null_terminated_string.characters(), 0755) < 0)
+    auto rc = mkdirat(dir_fd, null_terminated_string.characters(), 0755);
+    if (rc < 0)
```

Comment:
Not sure how this change is related; it also does not reflect the way you deal with error codes in a later commit (e.g. `if (fstat(...) < 0)`)